[CI] The latest mypy fails on current master

Mypy recently released a new 0.990 version that appears to be failing due to its internal error on modin's master.
```Shell
$ pip install mypy==0.990
$ mypy --config-file mypy.ini --show-traceback
/localdisk/dchigare/repos/modin/modin/experimental/pandas/numpy_wrap.py:118: error: INTERNAL ERROR -- Please try using mypy master on GitHub:
https://mypy.readthedocs.io/en/stable/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 0.990
Traceback (most recent call last):
AssertionError: Error checking function redefinition FuncDef:118
```

The issue is already reported to mypy python/mypy#14027 but there's yet neither a fix nor an upcoming release to resolve it. So for now we should pin mypy to the lower version in order to make our CI work.

--- a/environment-dev.yml
+++ b/environment-dev.yml
@@ -36,7 +36,7 @@ dependencies:
   - scikit-learn
   - pymssql
   - psycopg2
-  - mypy
+  - mypy<0.990
   - pandas-stubs
   - fastparquet
   - pip:

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -47,6 +47,6 @@ flake8-print
 numpydoc==1.1.0
 # experimental version of fuzzydata requires at least 0.0.6 to successfully resolve all dependencies
 fuzzydata>=0.0.6
-mypy
+mypy<0.990
 pandas-stubs
 fastparquet

